### PR TITLE
W-13789629 create pdk project gr

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 1]
```diff
@@ -2,4 +2,123 @@
...
+== Create a New Policy Project
+
+To create a custom policy project, run the PDK CLI `create` command, replacing `<my-custom-policy>` with the name of your new policy:
+
+[source,ssh]
+----
+anypoint-cli-v4 pdk policy-project create --name <my-custom-policy>
+----
+
+Running the `create` command creates a custom policy project with the specified name. The project appears in a new directory or a directory specified using the `--output-di` flag. The following tutorials refer to this as your project's root directory.
```
Comment (nledesma) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
Running the `create` command creates a custom policy project with the specified name. The project appears in a new directory or a directory specified using the `--output-dir` flag. The following tutorials refer to this as your project's root directory.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 2]
```diff
@@ -2,4 +2,123 @@
...
+
+The `assetID` of the new policy is the policy's name in lower case with the spaces amended together with a `-`. For example, a policy with the name `My Header Injection Policy` asstID is `my-header-injection-policy`.
```
Comment (nledesma) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
The `assetID` of the new policy is the policy's name in lower case with the spaces amended together with a `-`. For example, a policy with the name `My Header Injection Policy`, `assetID` is `my-header-injection-policy`. 
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 3]
```diff
@@ -2,4 +2,123 @@
...
+[source,ssh]
+----
+anypoint-cli-v4 pdk policy-project create --name <my-custom-policy>
+----
+
+Running the `create` command creates a custom policy project with the specified name. The project appears in a new directory or a directory specified using the `--output-di` flag. The following tutorials refer to this as your project's root directory.
+
+The `assetID` of the new policy is the policy's name in lower case with the spaces amended together with a `-`. For example, a policy with the name `My Header Injection Policy` asstID is `my-header-injection-policy`. 
+
+The `groupID` of the new policy is `groupID` of the organization configured in Anypoint CLI.
```
Comment (nledesma) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]: I would say:
The `groupID` of the new policy is `ID` of the organization configured in Anypoint CLI.

GroupID is an Exchange concept, which in most cases uses the ID of the organization that owns the asset.

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 4]
```diff
@@ -2,4 +2,123 @@
...
+
+The `definition` directory contains the `gcl.yaml` file that defines the policy's metadata and the configuration properties.
+
+To configure the `gcl.yaml` file, see xref:policies-pdk-create-schema-definition.adoc[].
+
+=== Cargo.toml
+
+Custom policies for Flex Gateway are developed in the Rust Programming language and compiled to a WebAssembly binary. 
+
+The `Cargo.toml` is Rust project manifest. Its `[package]` section contains the basic information of the policy implementation. The version specified in `Cargo.toml` matches the policy asset version. You can manage the policy version by editing the version number in `Cargo.toml`. The version follows the Semantic Versionsing (SemVer) format: `<major-version>.<minor-version>.<patch-version>`.
```
Comment (nledesma) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]: > The version follows the Semantic Versionsing (SemVer) format: `<major-version>.<minor-version>.<patch-version>`

This is not entirely accurate. SemVer permits some formats we do not support, for example: `1.0.1-integration`. The previous version is valid in SemVer, but not valid for us. We only admit the 3 numbered parts of the version: `1.0.1`, `1.2.3`, etc... So we accept a subset of SemVer format.

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 5]
```diff
@@ -2,4 +2,123 @@
...
+=== Makefile
+
+This file contains all the callable actions, `make` commands, required for the policy development lifecycle:
+
+* `make setup`: Installs the PDK internal dependencies for the rest of the Makefile goals. 
++
+You ran this command in <<setup-a-new-policy-project>>.
+* `make build-asset-files`: Generates all the policy asset files required to build, execute, and publish the policy. This command also updates the `config.rs` source code file with the latest configurations defined in the `gcl.yaml` file. 
++
+For more information about compiling a policy, see xref:policies-pdk-create-schema-definition.ado[].
```
Comment (nledesma) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]: This text is out of sync with the cited link

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 6]
```diff
@@ -2,4 +2,122 @@
...
+To create a custom policy project, run the PDK CLI `create` command, replacing `<my-custom-policy>` with the name of your new policy:
+
+[source,ssh]
+----
+anypoint-cli-v4 pdk policy-project create --name <my-custom-policy>
+----
+
+Running the `create` command creates a custom policy project with the specified name. The project appears in a new directory or a directory specified using the `--output-dir` flag. The following tutorials refer to this as your project's root directory.
+
+The `assetID` of the new policy is the policy's name in lower case with the spaces amended together with a `-`. For example, a policy with the name `My Header Injection Policy` has the `asstID` of `my-header-injection-policy`.
```
Comment (nledesma) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]: there's still a typo here: "asstID" => "assetID"

**Review by nledesma — APPROVED** [review]
LGTM

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 7]
```diff
@@ -2,4 +2,143 @@
...
+[source,ssh]
+----
+anypoint-cli-v4 pdk policy-project create --name <my-custom-policy>
+----
+
+Running the `create` command creates a custom policy project with the specified name. The project appears in a new directory or a directory specified using the `--output-dir` flag. The following tutorials refer to this as your project's root directory.
+
+The `assetID` of the new policy is the policy's name in lower case with the spaces amended together with a `-`. For example, a policy with the name `My Header Injection Policy` has the `assetID` of `my-header-injection-policy`. 
+
+The `groupID` of the new policy is ID of the organization configured in Anypoint CLI.
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
The `groupID` of the new policy is the ID of the organization configured in Anypoint CLI.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 8]
```diff
@@ -2,4 +2,143 @@
...
+
+Additionally, the `create` command supports the following flags:
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
The `create` command supports the following flags:
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 9]
```diff
@@ -2,4 +2,143 @@
...
+
+[%header%autowidth.spread,cols="a,a,a,a"]
+|===
+| Flag | Default Value | Required | Description
+| `--name=<value>` or `-n` | N/A | Yes | Name of the new policy
+| `--description=<value>` or `-d` | N/A | No | Description of the new policy
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
| `--name=<value>` or `-n` | N/A | Yes | Name of the new policy.
| `--description=<value>` or `-d` | N/A | No | Description of the new policy.
```
For consistency with the other items in the table.

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 10]
```diff
@@ -2,4 +2,143 @@
...
+| `--output-dir=<value>` of `-o` | `assetID` | No | Directory the new policy is created in. +
+If the directory does not exist, it is created. If no directory is specified, the directory name is the policy `assetID`.
+| `--version=<value>` | `1.0.0` | No | Version of the new policy. +
+The version format is `<major-release>.<minor-release>.<patch-release>`.
+|===
+
+== Setup the PDK Build Environment
+
+If you haven't previously setup your PDK build environment, you must run the `setup` command to install the PDK project's dependencies.
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
If you haven't set up your PDK build environment, run the `setup` command to install the PDK project's dependencies.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 11]
```diff
@@ -2,4 +2,143 @@
...
+
+To setup your PDK environment:
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
To set up your PDK environment:
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 12]
```diff
@@ -2,4 +2,143 @@
...
+
+. Execute the following `curl` command to retrive your connected app bearer token. Replace your connected app values:
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
. Execute the following `curl` command to retrieve your Connected App bearer token. Replace your Connected App values:
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 13]
```diff
@@ -2,4 +2,143 @@
...
+. Save the returned `access_token`.
+
+. Run the `setup` command:
++
+[source,ssh]
+----
+make setup
+----
+
+. Enter an `username` value.
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
. Enter a `username` value.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 14]
```diff
@@ -2,4 +2,143 @@
...
+├─ Cargo.toml   # Rust language standard file that contains the metadata required to build the policy
+└─.project.yaml # Project manifest used by PDK
+----
+
+=== target/ and definition/target/
+
+There are two separate `target` directories:
+
+* `definition/target/`: contains the policy asset definition published to exchange
+* `target/`: Contains the files for the Policy asset implementation and the output of the Rust compilation process
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
* `definition/target/`: Contains the policy asset definition published to Exchange.
* `target/`: Contains the files for the Policy asset implementation and the output of the Rust compilation process.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 15]
```diff
@@ -2,4 +2,143 @@
...
+
+The `definition` directory contains the `gcl.yaml` file that defines the policy's metadata and the configuration properties.
+
+To configure the `gcl.yaml` file, see xref:policies-pdk-create-schema-definition.adoc[].
+
+=== Cargo.toml
+
+Custom policies for Flex Gateway are developed in the Rust Programming language and compiled to a WebAssembly binary. 
+
+The `Cargo.toml` is Rust project manifest. Its `[package]` section contains the basic information of the policy implementation. The version specified in `Cargo.toml` matches the policy asset version. You can manage the policy version by editing the version number in `Cargo.toml`. The version follows the format `<major-version>.<minor-version>.<patch-version>`.
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
The `Cargo.toml` is the Rust project manifest. Its `[package]` section contains the basic information on the policy implementation. The version specified in `Cargo.toml` matches the policy asset version. You can manage the policy version by editing the version number in `Cargo.toml`. The version follows the format `<major-version>.<minor-version>.<patch-version>`.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 16]
```diff
@@ -2,4 +2,143 @@
...
+
+=== src/
+
+The `src` directory contains the policy's source code:
+
+* `lib.rs`: Contains a set of methods that implement the behavior of the policy. These methods are the API our development kit offers to implement custom policies behavior. 
+
+* `generated/config.rs` and `generated/mod.rs`: Are auto generated modules that make the policy configuration, as defined in the `gcl.yaml`, available for use in the lib.rs and any of the policy source files.
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
* `generated/config.rs` and `generated/mod.rs`: Auto-generated modules that make the policy configuration, as defined in the `gcl.yaml`, available for use in the `lib.rs` and any of the policy source files.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 17]
```diff
@@ -2,4 +2,143 @@
...
+
+=== Makefile
+
+This file contains all the callable actions, `make` commands, required for the policy development lifecycle. Execute all `make` commands at the root level of you root directory:
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
This file contains all the callable actions, `make` commands, required for the policy development lifecycle. Execute all `make` commands at the root level of your root directory:
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 18]
```diff
@@ -2,4 +2,143 @@
...
+
+* `make setup`: Installs the PDK internal dependencies for the rest of the Makefile goals. 
++
+You ran this command in <<setup-a-new-policy-project>>.
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
You run this command in <<setup-a-new-policy-project>>.
```
Please check if this is correct

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 19]
```diff
@@ -2,4 +2,143 @@
...
+* `make build-asset-files`: Generates all the policy asset files required to build, execute, and publish the policy. This command also updates the `config.rs` source code file with the latest configurations defined in the `gcl.yaml` file. 
++
+For more information about compiling a policy, see xref:policies-pdk-compile-policies.adoc[].
++
+* `make build`: Runs the `make build-asset-files` command and compiles the WebAssembly binary of the policy. 
++
+For more information about compiling a policy, see xref:policies-pdk-compile-policies.adoc[]. 
+* `make run`: Provides a simple way to execute the current build of the policy in Docker containerized environment.
```
Comment (Cristian-Venticinque) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
* `make run`: Provides a simple way to execute the current build of the policy in a Docker containerized environment. 
```

**Review by Cristian-Venticinque — APPROVED** [review]
Please check comments before merging

File: gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc (hunk 15 above)
Comment (glenn-rodgers-sf) [reply gateway/1.5/modules/ROOT/pages/policies-pdk-create-project.adoc]: Changed 'on' to 'about'

**Pull request merged**